packages/smooth_app/lib/cards/category_cards/svg_safe_network.dart

```diff
@@ -0,0 +1,94 @@
+import 'dart:ui' as ui;
+
+import 'package:flutter/material.dart';
+import 'package:flutter_svg/flutter_svg.dart';
+import 'package:flutter_svg/svg.dart';
+import 'package:http/http.dart' as http;
+import 'package:smooth_app/cards/category_cards/asset_cache_helper.dart';
+import 'package:smooth_app/cards/category_cards/svg_async_asset.dart';
+import 'package:smooth_app/cards/category_cards/svg_cache.dart';
+import 'package:smooth_app/services/smooth_services.dart';
+
+/// Widget with async load of SVG network file.
+///
+/// We could use SvgPicture.network, but it sends tons of errors if there in no
+/// internet connection. That's why we download the data ourselves.
+class SvgSafeNetwork extends StatefulWidget {
+  const SvgSafeNetwork(this.helper);
+
+  final AssetCacheHelper helper;
+
+  @override
+  State<SvgSafeNetwork> createState() => _SvgSafeNetworkState();
+}
+
+class _SvgSafeNetworkState extends State<SvgSafeNetwork> {
+  late final Future<String> _loading = _load();
+
+  String get _url => widget.helper.url;
+
+  Future<String> _load() async {
+    Map<String, String> cache = _getSafeCache();
+    String? cached = cache[_url];
+    if (cached != null) {
+      return cached;
+    }
+    final http.Response response = await http.get(Uri.parse(_url));
+    // in case we get gc'ed while await'ing
```

Comment:
What is gc'ed

Google: Global Citizenship Education

🤔😄

Reply:
Oh, `gc` is the Garbage Collector. In Java there a `gc()` method that you can explicitly call if needed.
I'll edit the comment for non Java developers.